mod-clans/src/main/java/gg/solarmc/loader/clans/ClanDataObject.java

```diff
@@ -0,0 +1,132 @@
+/*
+ *
+ *  * dataloader
+ *  * Copyright © 2021 SolarMC Developers
+ *  *
+ *  * dataloader is free software: you can redistribute it and/or modify
+ *  * it under the terms of the GNU Affero General Public License as
+ *  * published by the Free Software Foundation, either version 3 of the
+ *  * License, or (at your option) any later version.
+ *  *
+ *  * dataloader is distributed in the hope that it will be useful,
+ *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  * GNU Affero General Public License for more details.
+ *  *
+ *  * You should have received a copy of the GNU Affero General Public License
+ *  * along with dataloader. If not, see <https://www.gnu.org/licenses/>
+ *  * and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package gg.solarmc.loader.clans;/*
+ *
+ *  * dataloader
+ *  * Copyright © 2021 SolarMC Developers
+ *  *
+ *  * dataloader is free software: you can redistribute it and/or modify
+ *  * it under the terms of the GNU Affero General Public License as
+ *  * published by the Free Software Foundation, either version 3 of the
+ *  * License, or (at your option) any later version.
+ *  *
+ *  * dataloader is distributed in the hope that it will be useful,
+ *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  * GNU Affero General Public License for more details.
+ *  *
+ *  * You should have received a copy of the GNU Affero General Public License
+ *  * along with dataloader. If not, see <https://www.gnu.org/licenses/>
+ *  * and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+import gg.solarmc.loader.Transaction;
+import gg.solarmc.loader.data.DataObject;
+import gg.solarmc.loader.schema.tables.records.ClansClanMembershipRecord;
+import org.jooq.DSLContext;
+
+import java.util.Optional;
+
+import static gg.solarmc.loader.schema.tables.ClansClanMembership.CLANS_CLAN_MEMBERSHIP;
+
+
+/**
+ * Note to aurium - you don't have to put anything into the database when setting
+ * the cached clan *because this doesn't exist in the database*
+ *
+ * (gg.solarmc.loader.clans.Clan membership is stored in the clan membership table, not alongside the dataobject,
+ * and furthermore this dataobject isn't stored in a table)
+ */
+public class ClanDataObject implements DataObject {
+
+    private final int userId;
+    private final ClanManager manager;
+
+    private volatile Clan cachedClan; //nullable
+
+    public ClanDataObject(int userId, Clan clan, ClanManager manager) {
```

Comment:
Did you forget to set the cachedClan to clan? Also, this constructor should be package-private

Reply:
Fixed